openpype/hosts/houdini/api/lib.py

```diff
@@ -649,3 +655,71 @@ def get_color_management_preferences():
         "display": hou.Color.ocio_defaultDisplay(),
         "view": hou.Color.ocio_defaultView()
     }
+
+
+def publisher_show_and_publish():
+    """Open publisher window and trigger publishing action."""
+
+    main_window = get_main_window()
+    publisher_window = get_tool_by_name(
+        tool_name="publisher",
+        parent=main_window
+    )
+
+    publisher_window.set_current_tab("publish")
+    publisher_window.make_sure_is_visible()
+    publisher_window._reset_on_show = False
+
+    publisher_window._controller.reset()
+    publisher_window._controller.publish()
+
+
+def self_publish():
+    """Self publish from ROP nodes."""
+
+    current_node = hou.node(".").path()
+
+    host = registered_host()
+    context = CreateContext(host, reset=True)
+
+    for instance in context.instances:
```

Comment:
This loop won't disable other instances, so if will publish more than just selected instance, or?

Reply:
It's marked as resolved, but nothing changed?

Reply:
omg, I changed it locally but I didn't push my commit sorry

Reply:
what about these?
```
if not node_path:
    continue
```

Reply:
There's an instance that hasn't a `node_path`, this `instance.data.get("instance_node")`  equals `None`
so I skipped it as I deactivate instances using `node_path`

Reply:
What @iLLiCiTiT is trying to say is those nodes without `instance_node`, like the workfile instance will not be deactivated even though the user expects likely only that single instance they chose to be publishes.

I agree with @iLLiCiTiT that we're better off always setting active = False unless it's the instance we want.

Reply:
it didn't work until I made this 
```python
active = current_node == node_path
instance["active"] = active
hou.node(node_path).parm("active").set(active)
```